notebook/services/contents/manager.py

```diff
@@ -440,10 +440,15 @@ def check_and_sign(self, nb, path=''):
         path : string
             The notebook's path (for logging)
         """
+        # Warning: The check_cells method removes the 
+        # "trusted"" key from the cells' metadata 
         if self.notary.check_cells(nb):
-            self.notary.sign(nb)
+            # We're checking the signature *after* the
+            # trusted keys have been removed
+            if not self.notary.check_signature(nb):
```

Comment:
@minrk The way ``store_signature`` for the signature store is implemented, the signatures _don't_ get updated. 
I logged an issue with nbformat, let me know if I'm wrong [here](https://github.com/jupyter/nbformat/issues/99)